Specifying the KRMx01 machine and STM32F4-Discovery controller.
The limit switch debounce logic makes them active high.
The KRMx01 is a XYZY axis machine.
--- grbl-master/config.h	2014-03-14 07:50:36.000000000 -0700
+++ grbl-master/config.h	2014-08-17 08:24:50.000000000 -0700
@@ -29,14 +29,14 @@
 #define config_h
 
 // Default settings. Used when resetting EEPROM. Change to desired name in defaults.h
-#define DEFAULTS_GENERIC
+#define DEFAULTS_KRMX01
 
 // Serial baud rate
 #define BAUD_RATE 9600
 
 // Default pin mappings. Grbl officially supports the Arduino Uno only. Other processor types
 // may exist from user-supplied templates or directly user-defined in pin_map.h
-#define PIN_MAP_ARDUINO_UNO
+#define PIN_MAP_STM32F4_DISCOVERY
 
 // Define runtime command special characters. These characters are 'picked-off' directly from the
 // serial read data stream and are not passed to the grbl line execution parser. Select characters
@@ -169,13 +169,17 @@
 // NOTE: Uncomment to enable. The recommended delay must be > 3us and the total step pulse
 // time, which includes the Grbl settings pulse microseconds, must not exceed 127us. Reported
 // successful values for certain setups have ranged from 10 to 20us.
-// #define STEP_PULSE_DELAY 10 // Step pulse delay in microseconds. Default disabled.
+#define STEP_PULSE_DELAY 5 // Step pulse delay in microseconds. Default disabled.
 
 // Uncomment the following define if you are using hardware that drives high when your limits
 // are reached. You will need to ensure that you have appropriate pull-down resistors on the
 // limit switch input pins, or that your hardware drives the pins low when they are open (non-
 // triggered).
-// #define LIMIT_SWITCHES_ACTIVE_HIGH
+#define LIMIT_SWITCHES_ACTIVE_HIGH
+
+// Uncomment to make the A-axis of a 4-axis/XYZA controller match the Y axis.
+// That is: Two steppers are used to drive the Y-axis.
+#define AXES_XYZY
 
 // ---------------------------------------------------------------------------------------
 
